library/extractor/src/main/java/com/google/android/exoplayer2/extractor/ogg/VorbisReader.java

```diff
@@ -140,7 +144,11 @@ protected boolean readHeaders(ParsableByteArray packet, long position, SetupData
     }
 
     if (commentHeader == null) {
-      commentHeader = VorbisUtil.readVorbisCommentHeader(scratch);
+      // The comment header can be in a future page
+      if (VorbisUtil.verifyVorbisHeaderCapturePattern(/* headerType= */ 0x03, scratch, /* quiet= */ true)) {
+        scratch.setPosition(scratch.getPosition() - 7);
+        commentHeader = VorbisUtil.readVorbisCommentHeader(scratch);
+      }
```

Comment:
Based on the [spec](https://www.xiph.org/vorbis/doc/Vorbis_I_spec.html#x1-610004.2), the Ogg stream consists of three header packets which are in order. 

Do we need to do this also for `vorbisIdHeader`? Based on my understanding, I think we skip pages not belonging to vorbis stream in `OggPageHeader` and we don't need to do any changes here. WDYT?

Reply:
Yes the page is skipped before if needed for the vorbisIdHeader so need for change. If it's not present at that moment it's an error.

Previously it was assumed that the comment was in the same page and directly read, but the spec does not enforce that and I had multiple files during my tests that had the comment on the next page.

Reply:
The sample file from the OG issue have the case. As explained you mix packet and page. The packets are put in pages. But the pages can be anysize and interleaved specially when interleaving for those multi tracks.

So you have [Stream1-Page1][Header][Packet1][Stream2-Page1][Header][Packet1][Stream1-Page2][Header][Packet2][Stream2-Page2][Header][Packet2]

Packet 1 and packet 2 are in order and adjoined as the spec define. But the actual bytes are not at all, there's the other stream + the page header between them.